### ISSUE TITLE
ENUM values fail to load in some circumstances

### ISSUE THREAD

**Issue by cookieguru** [HeidiSQL/HeidiSQL#1208]
In some circumstances, joining a table causes ENUM values fail to load

**To Reproduce**
1. Run query with a join (see below) and see results
2. Click dropdown for enum column
3. See no values
4. Run query without join and see results
5. Click dropdown for enum column
6. See values

**Expected behavior**
Values for enum column always display

**Screenshots**
With join:
![screenshot of dropdown using query with join](https://user-images.githubusercontent.com/1888809/98451881-f50d3e80-20fe-11eb-91b1-c95b51170c55.png)

Without join:
![screenshot of dropdown using query without join](https://user-images.githubusercontent.com/1888809/98451866-e32b9b80-20fe-11eb-8294-3772f7c799d2.png)

**Environment:**
 - OS: Windows 10
 - HeidiSQL version: 11.0.0.6116
 - Database type and version: MySQL 5.7.32

**Table structure:**
(several columns omitted for brevity)
```sql
CREATE TABLE `episodes` (
	`id` CHAR(24) NOT NULL DEFAULT '' COLLATE 'utf8mb4_unicode_ci',
	`season` CHAR(24) NOT NULL DEFAULT '' COLLATE 'utf8mb4_unicode_ci',
	`downloaded` ENUM('Y','N') NOT NULL DEFAULT 'N' COLLATE 'utf8mb4_unicode_ci',
	PRIMARY KEY (`id`) USING BTREE,
	INDEX `FK_episodes_seasons` (`season`) USING BTREE
)
COLLATE='utf8mb4_unicode_ci'
ENGINE=InnoDB
;
CREATE TABLE `seasons` (
	`id` VARCHAR(24) NOT NULL COLLATE 'utf8mb4_unicode_ci',
	`number` VARCHAR(50) NOT NULL COLLATE 'utf8mb4_unicode_ci',
	PRIMARY KEY (`id`) USING BTREE,
)
COLLATE='utf8mb4_unicode_ci'
ENGINE=InnoDB
;
```

**Queries**
Without join:
```sql
SELECT e.id, e.name, e.downloaded FROM episodes e
WHERE e.`show` = 'abc123' AND downloaded = 'N'
```
With join:
```sql
SELECT e.id, s.number, e.number, e.name, downloaded FROM episodes e
LEFT JOIN seasons s ON e.season = s.id
WHERE e.`show` = 'abc123' AND downloaded = 'N'
ORDER BY s.number, e.number
```

**Comment by ansgarbecker** [comment]
Yes, a result with multiple tables leads HeidiSQL to stop looking for column details like ENUM values. I just examined the internals here, with a code path through `TDBQuery.PrepareEditing` and `TDBQuery.PrepareColumnAttributes`. The internal debug message _More than one table involved._ is thrown by design, so I'd rather label this an enhancement, not a bug. But yes, this should be supported.

Please note that even when I have a fix for this, and ENUM values are shown in the drop-down, the result will still be non-editable. This is for safety reasons, as multiple tables in a result can result in wrong `UPDATE` queries.

**Comment by cookieguru** [comment]
If the value isn't editable, why is the dropdown interactable at all?  Or could editing be supported if a join occurs but only the columns of one table are selected?

**Comment by ansgarbecker** [comment]
> If the value isn't editable, why is the dropdown interactable at all?

Because many users wanted editable grid cells, because especially the text editor has a better readability.

> Or could editing be supported if a join occurs but only the columns of one table are selected?

That is already the case now, see here:

![grafik](https://user-images.githubusercontent.com/7986591/98634276-5281d500-2323-11eb-8fa2-0273317486e2.png)
